rust/nixops4-resource/src/framework.rs

```diff
@@ -44,79 +52,121 @@ fn write_response<W: std::io::Write>(mut out: W, resp: &v0::Response) -> Result<
     out.flush().context("flushing response")
 }
 
+#[derive(Default)]
+pub struct ContentLengthCodec {
+    content_length: Option<usize>,
+}
+
+impl ContentLengthCodec {
+    fn decode_headers(&mut self, src: &mut BytesMut) -> Result<(), Error> {
+        if let Some(pos) = src.windows(2).position(|w| w == b"\n\n") {
```

Comment:
Ah this is why the test isn't working. Inspired by the HTTP spec, this is supposed to be `\r\n\r\n`. I had removed the carriage returns to make some manual testing easier. We could do something else, or the content length framing might be overkill instead of just single-line requests.
```suggestion
        if let Some(pos) = src.windows(2).position(|w| w == b"\r\n\r\n") {
```